e3sm_diags/driver/utils/regrid.py

```diff
@@ -388,24 +364,30 @@ def align_grids_to_lower_res(
     Tuple[xr.Dataset, xr.Dataset]
         A tuple of both DataArrays regridded to the lower resolution of the two.
 
-    Notes
-    -----
-    Replaces `e3sm_diags.driver.utils.general.regrid_to_lower_res`.
-
     References
     ----------
     .. [1] https://xcdat.readthedocs.io/en/stable/generated/xarray.Dataset.regridder.horizontal.html
     """
-    # TODO: Accept "esmf" as `tool` value for now because `CoreParameter`
-    # defines `self.regrid_tool="esmf"` by default and
-    # `e3sm_diags.driver.utils.general.regrid_to_lower_res()` accepts "esmf".
-    # Once this function is deprecated, we can remove "esmf" as an option here
-    # and update `CoreParameter.regrid_tool` to "xesmf"`.
+    # TODO: Remove this conditional once "esmf" references are updated to
+    # "xesmf" throughout the codebase.
     if tool == "esmf":
         tool = "xesmf"
 
-    ds_a_new = _drop_unused_ilev_axis(ds_a)
-    ds_b_new = _drop_unused_ilev_axis(ds_b)
+    # TODO: Remove this conditional once "conservative" references are updated
+    # to "conservative_normed" throughout the codebase.
+    # NOTE: this is equivalent to "conservative" in cdms2 ESMF. If
+    # "conservative" is chosen, it is updated to "conservative_normed". This
+    # logic can be removed once the CoreParameter.regrid_method default
+    # value is updated to "conservative_normed" and all sets have been
+    # refactored to use this function.
+    if method == "conservative":
+        method = "conservative_normed"
```

Comment:
Related to https://github.com/E3SM-Project/e3sm_diags/issues/938#issuecomment-2664082359

I think I forgot to add this conditional in `_align_grids_to_lower_res()`, which resulted in the abnormal latitude edge in the polar plot [here](https://github.com/E3SM-Project/e3sm_diags/pull/931#issuecomment-2652403024). This fixes that issue and hopefully more diffs between "Model - Observation" plots.

Reply:
@chengzhuzhang I just did a run with the specific variables from #930 that have differences due to regridding. This change doesn't seem to affect the plot differences for these cases. The regression testing notebooks show the same differences between `*_diff.nc` files and the plot diffs look similar.

I think this is the new, expected result as far as I can tell (did not try the bounds dimension name alignment fix from #938).

Reply:
This fix does work with the `polar` plots though.

Reply:
@tomvothecoder see comment here: https://github.com/E3SM-Project/e3sm_diags/issues/938#issuecomment-2666940058. It looks like changing to `convervative_normed` has little impact, I'm interested to look at the plots and tables. Could you point me to the urls of both runs?

Reply:
> Could you point me to the urls of both runs?

- Latest `main` with `conservative`: https://web.lcrc.anl.gov/public/e3sm/cdat-migration-fy24/25-02-04-branch-930-zppy-diffs/viewer/index.html
- This branch with `conservative_normed`: https://web.lcrc.anl.gov/public/e3sm/cdat-migration-fy24/25-02-18-branch-940-xesmf-diffs/viewer/index.html

Reply:
Thanks for the links! I think we should use `conservative_normed` to replace `bilinear`. Now there is a caveat: [here](https://web.lcrc.anl.gov/public/e3sm/cdat-migration-fy24/25-02-18-branch-940-xesmf-diffs-no-bilinear/viewer/lat_lon/era5/trefht-land-era5/ann.html), look at the diff plot, there is a gap along the coastal line, in conservative_normed. And I realize that, part of the reason is that we are using a very coarse resolution file for land-sea mask. I'm creating a new land-sea mask to see if it helps.